tasks/test_e2e.yml

```diff
@@ -0,0 +1,86 @@
+      register: checkmyconf
+      failed_when:
+        - checkmyconf.rc !=0 
+    
+    - name: "Capture collectd_cpu_percent"
+      shell: /usr/bin/curl -k http://{{ prom_url.stdout_lines[0] }}/api/v1/query? --data-urlencode 'query=collectd_cpu_percent {plugin_instance="0",type_instance="user",service="default-cloud1-coll-meter-smartgateway"}[1m]' > /tmp/query_collectd_cpu_percent
+      register: checkmyconf
+      failed_when:
+        - checkmyconf.rc !=0
+    
+    - name: "Capture collectd_iinterface_tx_total"
+      shell: /usr/bin/curl -k http://{{ prom_url.stdout_lines[0] }}/api/v1/query? --data-urlencode 'query=collectd_interface_if_packets_tx_total {type_instance="base"}[1m]' > /tmp/query_collectd_interface_tx_total
+      register: checkmyconf
+      failed_when:
+        - checkmyconf.rc !=0
+    
+    - name: "Capture collectd_memory"
+      shell: /usr/bin/curl -k curl -k http://{{ prom_url.stdout_lines[0] }}/api/v1/query? --data-urlencode 'query=collectd_memory {plugin_instance="base"}[1m]' > /tmp/query_collectd_memory
+      register: checkmyconf
+      failed_when:
+        - checkmyconf.rc !=0
+
+    - name: "Capture collectd_load_longterm"
+      shell: /usr/bin/curl -k curl -k http://{{ prom_url.stdout_lines[0] }}/api/v1/query? --data-urlencode 'query=collectd_load_longterm {plugin_instance="base"}[1m]' > /tmp/query_load_longterm
+      register: checkmyconf
+      failed_when:
+        - checkmyconf.rc !=0
+
+    - name: "[Test] Check whether ceph bytes metrics received in prometheus for any ceph nodes"
+      command: egrep 'ceph' /tmp/query_ceph_ceph_bytes
+      register: checkmyconf
+      changed_when: no
+      failed_when:
+        - checkmyconf.rc != 0
+    - debug:
+        msg="COLLECTD CEPH BYTES METRRICS RECEIVED FOR ANY CEPH NODES"
+
+    - name: "[Test] Check whether collectd interface TX total metrics received in prometheus for all overcloud nodes"
+      command: egrep 'controller-0|controller-1|controller-2|compute-0|compute-1|ceph-0' /tmp/query_collectd_interface_tx_total
+      register: checkmyconf
+      changed_when: no
+      failed_when:
+        - checkmyconf.rc != 0
+    - debug:
+        msg="COLLECTD INTERFACE TOTAL TX  METRICS RECEIVED FOR ANY OVERCLOUD NODES"
+
+    - name: "[Test] Check whether collectd cpu metrics log created and received in Prometheus"
+      command: egrep 'controller-0|controller-1|controller-2|compute-0|compute-1|ceph-0' /tmp/query_collectd_cpu_percent
+      register: checkmyconf
+      changed_when: no
+      failed_when:
+        - checkmyconf.rc != 0
+    - debug:
+        msg="COLLECTD CPU METRICS RECEIVED FOR ANY OVERCLOUD NODES"
+
+
+    - name: "[Test] Check whether collectd memory metrics log created and received in Prometheus"
+      command: egrep 'controller-0|controller-1|controller-2|compute-0|compute-1|ceph-0' /tmp/query_collectd_memory
+      register: checkmyconf
+      changed_when: no
+      failed_when:
+        - checkmyconf.rc != 0
+    - debug:
```

Comment:
This debug message will output regardless of whether the previous task failed or succeeded.
I would suggest either adding a `when` to this debug or splitting the previous task and doing the checking in an `[assert](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/assert_module.html)` task, using a success_msg option

e.g.
```
   - name: "[Set-up] Check whether collectd memory metrics log created and received in prometheus"
      ...
      changed_when: no
    - name: "[Test] Check whether collectd memory metrics log created ans received in Prometheus"
      assert:
        that:
            - checkmyconf.rc == 0
        success_msg: "COLLECTD MEMORY METRICS RECEIVED FOR ANY OVERCLOUD NODES"

```

This is applicable for each test/debug task pair.
My concern is that outputting that debug message will cause confusion if it is always printed.

Another option is to remove the debug, but I suspect it was there for a reason.